### DISCUSSION TITLE
Ray vs Telescope

### DISCUSSION THREAD

**Discussion by amosmos** [spatie/ray#435] [Q&A] [answered]
Hi,

This app looks amazing!

I was wondering what is the difference between using Ray vs Telescope.
Am I right that they are they serving the same purpose?
If so what is the right way to consider using each one for a project?

THANKS!
Amos

**Comment by tpetry** [comment, marked as answer]
Telescope does have debugging functionality but ray has a lot more features. And ray is not limited to your laravel application, you can use from more languages and even in the javascript executed in your browser.

**Reply by amosmos** [reply]
Thanks. Does that means that Telescope and Ray serve the same purpose only with different features?

**Reply by fshequin** [reply]
No....different and typically different use cases...Ray detailed debugging of logic you are working on by giving you a super duper dd() in a separate app native to your computer. Telescope if HTML requests, higher level view of total working s of app....

**Reply by fshequin** [reply]
Ray becomes worth it when you are producing oodles of logic and need to test every little bit very carefully but faster because you do not have to change screens both can be up at the same time your app and your ray view of the varibles and objects that you've tagged to be reported out as they occur...

**Reply by fshequin** [reply]
vastly quicker verification of logic....less clicks
